src/TSHelper.ts

```diff
@@ -76,6 +76,10 @@ export class TSHelper {
         return extendedTypeNode && checker.getTypeAtLocation(extendedTypeNode);
     }
 
+    public static isAmbientModuleDeclaration(type: ts.Type): boolean {
+        return type.symbol && type.symbol.valueDeclaration.kind === ts.SyntaxKind.ModuleDeclaration;
```

Comment:
After some discussion in the discord we came to the conclusion that it would be a better solution to add a compiler directive `@noResolution` on the module, to minimize the risk of this breaking something.